framework/include/utils/MooseUtils.h

```diff
@@ -1065,6 +1065,25 @@ get(const std::shared_ptr<T> & s)
   return s.get();
 }
 
+// print a message from all processors (use only for debugging purposes)
+template <typename... Args>
+void
+parallelPrint(const Parallel::Communicator & comm, const std::string & title, Args &&... args)
```

Comment:
This came in handy for parallel debugging and beats `--keep-cout --redirect-stdout` plus `grep`, IMO

Reply:
`SerializerGuard` to cerr does this too :-)

Reply:
No, it does not.

Reply:
To clarify, if you use `std::cout` with a barrier you still need to supply `--keep-cout`, which gives you _all_ console output N times.

Reply:
oh... good point.